src/script/interpreter.cpp

```diff
@@ -2714,8 +2712,26 @@ static bool ExecuteWitnessScript(const Span<const valtype>& stack_span, const CS
     }
 
     // Run the script interpreter.
+    // If serror is nullprt, set it to freshly created instance
+    // Make sure to set the freshly created instance back to nullptr for memory consistency
+    bool is_serror_null = (serror == nullptr);
+    if (!serror) {
+        ScriptError err = SCRIPT_ERR_UNKNOWN_ERROR;
+        serror = &err;
+    }
     if (!EvalScript(stack, scriptPubKey, flags, checker, sigversion, execdata, serror)) return false;
 
+    // Set script success for early return. This skips the checks for CLEANSTACK, SCRIPT_ERR_EVAL_FALSE check.
```

Comment:
Also, I don't think you need to set `serror` back to null. I think you can just do
```
ScriptError _serror;
if (serror == nullptr) serror = &_serror;  // ELEMENTS: we need this error value even if the caller doesn't
```
and then leave the rest of the existing logic alone.
￼

Reply:
We never return this pointer. We return a boolean.